Lib/tempfile.py

```diff
@@ -822,8 +822,17 @@ def resetperms(path):
 
                     try:
                         _os.unlink(path)
+                    except IsADirectoryError:
+                        cls._rmtree(path, ignore_errors=ignore_errors)
                     # PermissionError is raised on FreeBSD for directories
-                    except (IsADirectoryError, PermissionError):
+                    except PermissionError:
+                        # On Windows, calling _rmtree again will raise
```

Comment:
Note that calling `unlink()` on a directory in Windows always raises `PermissionError`. Windows maps the NT status code `STATUS_FILE_IS_A_DIRECTORY` to the WinAPI error code `ERROR_ACCESS_DENIED`.

Reply:
> I think this would also happen on other OSes, not just Windows. The bug can only be reproduced on windows

I'm confused, should we continue the `_os.name == 'nt'` check then? I'm not worried about the additional performance hit from `isdir` (though I would like to avoid it). Correctness is more important. If this *does* happen on other OSes, then we should remove that OS check.

Reply:
We can remove the os check. It can happen on other OSes. With the check removed, the logic becomes "if we are not permitted to remove the file, return or reraise", which makes perfect sense.